src/test/java/Test40.java

```diff
@@ -0,0 +1,129 @@
+        WebElement signOutLink = driver.findElement(HomePage.HOME_MENU);
+        new WebDriverWait(driver, 10).until(ExpectedConditions.visibilityOf(signOutLink));
+        Assert.assertTrue(signOutLink.isDisplayed());
+    }
+
+    @Test
+    public void testTwoTest() {
+        WebElement loginButton = driver.findElement(LoginPage.LOGIN_BUTTON);
+        WebElement userNameInput = driver.findElement(LoginPage.USERNAME_INPUT);
+        WebElement passwordInput = driver.findElement(LoginPage.PASSWORD_INPUT);
+        new WebDriverWait(driver, 10).until(ExpectedConditions.visibilityOf(loginButton));
+        Assert.assertTrue(loginButton.isDisplayed());
+        userNameInput.sendKeys(LOGIN);
+        passwordInput.sendKeys(PASS);
+        Assert.assertEquals(userNameInput.getAttribute("value"), LOGIN);
+        Assert.assertEquals(passwordInput.getAttribute("value"), PASS);
+        driver.findElement(LoginPage.LOGIN_BUTTON).click();
+        WebElement signOutLink = driver.findElement(HomePage.HOME_MENU);
+        Assert.assertTrue(signOutLink.isDisplayed());
+        driver.findElement(HomePage.OFFICE_MENU).click();
+        WebElement searchInput = driver.findElement(OfficePage.SEARCH_INPUT);
+        new WebDriverWait(driver, 15, 2700).until(ExpectedConditions.visibilityOf(searchInput));
+        Assert.assertTrue(searchInput.isDisplayed());
+    }
+
+    @Test(dataProvider = "testData")
+    public void ddtTest(By locator) {
+        LoginPage.login(driver, LOGIN, PASS);
+        WebElement element = driver.findElement(locator);
+        new WebDriverWait(driver, 10).until(ExpectedConditions.visibilityOf(element));
+        Assert.assertTrue(element.isDisplayed());
+    }
+
+    @Test
+    public void iFrameTest() {
+        String word1 = "Hello ";
+        String word2 = "world!";
+        driver.get(IFramePage.I_FRAME_URL);
+        driver.switchTo().frame(IFramePage.FRAME_LOCATOR);
+        WebElement frameBody = driver.findElement(IFramePage.FRAME_BODY);
+        frameBody.clear();
+        frameBody.sendKeys(word1);
+        driver.switchTo().defaultContent();
+        driver.findElement(IFramePage.FRAME_BOLD_BUTTON).click();
+        driver.switchTo().frame(IFramePage.FRAME_LOCATOR);
+        frameBody.sendKeys(word2);
+        String frameText = frameBody.findElement(By.cssSelector("p")).getText();
+        String result = frameText.replaceAll("[\uFEFF]", "");
+        Assert.assertEquals(result, word1 + word2);
+        Assert.assertEquals(frameBody.findElement(By.cssSelector("strong")).getText(), word2);
+    }
+
+    @Test(dataProvider = "alertsTestData")
+    public void alertsTest(By locator, String expectedMessage, String resultMessage) {
+        driver.get(AlertsPage.ALERTS_URL);
+        WebElement resultText = driver.findElement(AlertsPage.RESULT_MESSAGE);
+        driver.findElement(locator).click();
+
+        Alert alert = driver.switchTo().alert();
+        Assert.assertEquals(alert.getText(), expectedMessage);
```

Comment:
It is better to use SoftAssert here or even split into 2 tests with single assertion each